cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -81,48 +58,30 @@ func CreateEssentialAddons(cfg *kubeadmapi.MasterConfiguration, client clientset
 		return fmt.Errorf("error when parsing kube-proxy configmap template: %v", err)
 	}
 
-	err = CreateKubeProxyAddon(proxyConfigMapBytes, proxyDaemonSetBytes, client)
-	if err != nil {
-		return err
-	}
-	fmt.Println("[addons] Applied essential addon: kube-proxy")
-
-	err = CreateKubeDNSAddon(dnsDeploymentBytes, dnsServiceBytes, client)
-	if err != nil {
+	if err = createKubeDNSAddon(dnsDeploymentBytes, dnsServiceBytes, client); err != nil {
 		return err
 	}
 	fmt.Println("[addons] Applied essential addon: kube-dns")
 	return nil
 }
 
-func CreateKubeProxyAddon(configMapBytes, daemonSetbytes []byte, client clientset.Interface) error {
-	kubeproxyConfigMap := &v1.ConfigMap{}
-	if err := kuberuntime.DecodeInto(api.Codecs.UniversalDecoder(), configMapBytes, kubeproxyConfigMap); err != nil {
-		return fmt.Errorf("unable to decode kube-proxy configmap %v", err)
-	}
-
-	if err := apiclientutil.CreateConfigMapIfNotExists(client, kubeproxyConfigMap); err != nil {
-		return err
-	}
-
-	kubeproxyDaemonSet := &extensions.DaemonSet{}
-	if err := kuberuntime.DecodeInto(api.Codecs.UniversalDecoder(), daemonSetbytes, kubeproxyDaemonSet); err != nil {
-		return fmt.Errorf("unable to decode kube-proxy daemonset %v", err)
+// CreateServiceAccount creates the necessary serviceaccounts that kubeadm uses/might use, if they don't already exist.
+func CreateServiceAccount(client clientset.Interface) error {
```

Comment:
I don't think this needs to be exposed separately.
Well, if we do that it's fine by me, but at least `EnsureDNSAddon` should call it.
(EnsureDNSAddon should not have dependencies)